x-pack/plugins/observability/public/utils/test_helper.tsx

```diff
@@ -62,3 +62,15 @@ export const render = (component: React.ReactNode) => {
     </IntlProvider>
   );
 };
+
+export function mockHook<
```

Comment:
A note on the generics:
`jest.FunctionPropertyNames<T>` returns a union type of the names of all the properties in `T` that are a function. 
I then use that union type to define the constraint on `T`, which says, for each value in this union type, make sure the type of those keys is AnyFunction. 
That in turn enables `U` to be safely defined as `ReturnType<T[K]>` since it knows that any valid `T[K]` will be a function so it will have a return type.